tests/security/SingleParticipantWithSecurity/SingleParticipantWithSecurity.cpp

```diff
@@ -0,0 +1,336 @@
+        status));
+      return;
+    }
+
+    std::cout << "SampleInfo.sample_rank = " << si.sample_rank << std::endl;
+    std::cout << "SampleInfo.instance_state = " << OpenDDS::DCPS::InstanceState::instance_state_string(si.instance_state) << std::endl;
+
+    if (si.valid_data) {
+
+      std::cout << "Message: subject    = " << message.subject.in() << std::endl
+        << "         subject_id = " << message.subject_id << std::endl
+        << "         from       = " << message.from.in() << std::endl
+        << "         count      = " << message.count << std::endl
+        << "         text       = " << message.text.in() << std::endl;
+
+      if (std::string("Comic Book Guy") != message.from.in() &&
+        std::string("OpenDDS-Java") != message.from.in()) {
+        std::cout << "ERROR: Invalid message.from" << std::endl;
+      }
+      if (std::string("Review") != message.subject.in()) {
+        std::cout << "ERROR: Invalid message.subject" << std::endl;
+      }
+      if (std::string("Worst. Movie. Ever.") != message.text.in()) {
+        std::cout << "ERROR: Invalid message.text" << std::endl;
+      }
+      if (message.subject_id != 99) {
+        std::cout << "ERROR: Invalid message.subject_id" << std::endl;
+      }
+    } else if (si.instance_state == DDS::NOT_ALIVE_DISPOSED_INSTANCE_STATE) {
+      ACE_DEBUG((LM_DEBUG, ACE_TEXT("%N:%l: INFO: instance is disposed\n")));
+    } else if (si.instance_state == DDS::NOT_ALIVE_NO_WRITERS_INSTANCE_STATE) {
+      ACE_DEBUG((LM_DEBUG, ACE_TEXT("%N:%l: INFO: instance is unregistered\n")));
+    } else {
+      ACE_ERROR((LM_ERROR,
+        ACE_TEXT("%N:%l: read()")
+        ACE_TEXT(" ERROR: unknown instance state: %d\n"),
+        si.instance_state));
+    }
+
+    return;
+  }
+  catch (const CORBA::Exception& e) {
+    e._tao_print_exception("Exception caught in on_data_available():");
+    ACE_OS::exit(EXIT_FAILURE);
+    return;
+  }
+}
+
+int ACE_TMAIN(int argc, ACE_TCHAR *argv[])
+{
+  int status = EXIT_SUCCESS;
+  DDS::DomainParticipantFactory_var dpf;
+  DDS::DomainParticipant_var participant;
+
+  try {
+
+    std::cout << "Starting publisher" << std::endl;
+    {
+      // Initialize DomainParticipantFactory
+      dpf = TheParticipantFactoryWithArgs(argc, argv);
+
+      std::cout << "Starting publisher with " << argc << " args" << std::endl;
+
+      DDS::DomainParticipantQos part_qos;
+      dpf->get_default_participant_qos(part_qos);
+
+      DDS::PropertySeq& props = part_qos.property.value;
+
+      // Determine the path to the keys
+      OPENDDS_STRING path_to_tests;
+      const char* dds_root = ACE_OS::getenv("DDS_ROOT");
+      if (dds_root && dds_root[0]) {
+        // Use DDS_ROOT in case we are one of the CMake tests
+        path_to_tests = OPENDDS_STRING("file:") + dds_root + "/tests/";
+      } else {
+        // Else if DDS_ROOT isn't defined try to do it relative to the traditional location
+        path_to_tests = "file:../../";
+      }
```

Comment:
```suggestion
      const OpenDDS::DCPS::String path_to_tests = "file:../../";
```

Reply:
Actually the paths should probably be rewritten since it was made for a test in `tests/DCPS`, not one in `test/security` that happens to work.